yente/search/mapping.py

```diff
@@ -91,12 +93,29 @@ def make_entity_mapping(schemata: Optional[Iterable[Schema]] = None) -> Dict[str
             # to facet on them.
             if prop.type.group is not None and not excluded:
                 copy_to.append(prop.type.group)
-            prop_mapping[name] = make_type_field(
-                prop.type,
-                copy_to=copy_to,
-                existing=prop_mapping.get(name, None),
+            prop_name_to_fields[name].append(
+                make_type_field(prop.type, copy_to=copy_to)
             )
 
+    prop_mapping: Dict[str, MappingProperty] = {}
+    for prop_name, fields in prop_name_to_fields.items():
+        merged_copy_to = list(
+            set(itertools.chain.from_iterable([f["copy_to"] for f in fields]))
+        )
+        text_fields = [f for f in fields if f["type"] == "text"]
+        keyword_fields = [f for f in fields if f["type"] == "keyword"]
+
+        # All fields within a group (text/keyword) are usually the same. We choose the
+        # last one cause that's what we used to do, but it really doesn't matter.
+        # keyword has precedence over text
+        # Currently, only authority causes a collision (some are string, some are entity)
+        selected_field = keyword_fields[-1] if keyword_fields else text_fields[-1]
+        # We merge the copy_to fields, just so that search still behaves as expected.
+        # even in the case of multiple fields with the same name.
+        selected_field["copy_to"] = merged_copy_to
```

Comment:
Just trying a different wording with a bit more precise terminology to see if I'm understanding correctly. No pressure to use this, but if I misunderstood, let me know please

```suggestion
        # All properties with the same name ought to map to the same field definition.
        # If a conflict occurs, we choose field type "keyword" over "text".
        # Currently, only the property "authority" has conflicts (some are string, some are entity)
        selected_field = keyword_fields[-1] if keyword_fields else text_fields[-1]
        # We merge the copy_to fields rather than choosing one value, just so that
        # queries work as expected in case one property maps to different copy_to than another
        # with the same field name.
        selected_field["copy_to"] = merged_copy_to
```